docs/usage/dns-autoscaling.md

```diff
@@ -0,0 +1,63 @@
+# DNS Autoscaling
+
+This is a short guide describing different options how to automatically scale CoreDNS in the shoot cluster.
+
+## Background
+
+Currently, Gardener uses CoreDNS as DNS server. Per default, it is installed as a deployment into the shoot cluster that is auto-scaled horizontally to cover for QPS-intensive applications. However, doing so does not seem to be enough to completely circumvent DNS bottlenecks such as:
+
+- Cloud provider limits for DNS lookups.
+- Unreliable UDP connections that forces a period of timeout in case packets are dropped.
+- Unnecessary node hopping since CoreDNS is not deployed on all nodes, and as a result DNS queries end-up traversing multiple nodes before reaching the destination server.
+- Inefficient load-balancing of services (e.g., round-robin might not be enough when using IPTables mode).
+- Overload of the CoreDNS replicas as the maximum amount of replicas is fixed.
+- and more ...
+
+As an alternative with extended configuration options, Gardener provides cluster-proportional autoscaling of CoreDNS. This guide focuses on the configuration of cluster-proportional autoscaling of CoreDNS and its advantages/disadvantages compared to the horizontal
+autoscaling.
+Please note that there is also the option to use a [node-local DNS cache](node-local-dns.md), which helps mitigate potential DNS bottlenecks (see [Trade-offs in conjunction with NodeLocalDNS](#trade-offs-in-conjunction-with-nodelocaldns) for considerations regarding using NodeLocalDNS together with one of the CoreDNS autoscaling approaches).
+
+## Configuring cluster-proportional DNS Autoscaling
+
+All that needs to be done to enable the usage of cluster-proportional autoscaling of CoreDNS is to set the corresponding option (`spec.systemComponents.coreDNS.autoscaling.mode`) in the `Shoot` resource to `cluster-proportional`:
+
+```yaml
+...
+spec:
+  ...
+  systemComponents:
+    coreDNS:
+      autoscaling:
+        mode: cluster-proportional
+...
+```
+
+To switch back to horizontal DNS autoscaling you can set the `spec.systemComponents.coreDNS.autoscaling.mode` to `horizontal` (or remove the `coreDNS` section).
+
+Once the cluster-proportional autoscaling of CoreDNS has been enabled and the Shoot cluster has been reconciled afterwards, a ConfigMap called `core-dns-autoscaler` will be created in the `kube-system` namespace with the default settings. The content will be similar to the following:
```

Comment:
Nit: Remove the `-` in `core-dns` since everything else is called `coredns*`.

Reply:
Renamed with latest commit.